setuptools_rust/setuptools_ext.py

```diff
@@ -108,6 +107,14 @@ def finalize_options(self):
                         self.distribution.entry_points["console_scripts"] = ep_scripts
 
                 bdist_wheel_base_class.finalize_options(self)
+
+            def get_tag(self):
+                python, abi, plat = super().get_tag()
+                universal2 = all(ext.universal2 for ext in self.distribution.rust_extensions) \
+                    if self.distribution.rust_extensions else False
+                if universal2 and plat.startswith("macosx_"):
+                    plat = "macosx_10_9_universal2"
```

Comment:
I am not sure if this is the right way to set the platform tag for universal2 wheel.

Reply:
It looks like `sysconfig.get_platform()` might return something with `universal2` in it, which could be better? https://github.com/pypa/setuptools/issues/2520

Reply:
That only returns `macosx-10.9-universal2` on macOS universal2 Python build, does not play well with cross compilation.

Reply:
It doesn't return `universal2` with Apple's bundled Python in Xcode which is universal2 build itself.

```bash
❯ python3
Python 3.8.2 (default, Dec 21 2020, 15:06:03)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.prefix
'/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8'
>>> import sysconfig; sysconfig.get_platform()
'macosx-10.14.6-arm64'

❯ file /Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/bin/python3.8
/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/bin/python3.8: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64]
/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/bin/python3.8 (for architecture x86_64):	Mach-O 64-bit executable x86_64
/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/bin/python3.8 (for architecture arm64):	Mach-O 64-bit executable arm64
```

Reply:
> It looks like `sysconfig.get_platform()` might return something with `universal2` in it, which could be better? [pypa/setuptools#2520](https://github.com/pypa/setuptools/issues/2520)

Sysconfig.get_platform() will return "universal2" as the "CPU architecture" when Python was build as a universal 2 binary using the configure script.  This is true for the (currently experimental) universal2 installer for Python 3.9.1 on Python.org, as wel as recent 3.10 alpha installers. 

The python included in Xcode does not return "universal2" in the platform tag, likely because it was build twice and lipo-d together.

My reason for filing #108 is to have a way to build universal2 wheels that match the python.org installers (including a deployment target of macOS 10.9 if that is at all possible using the default rust toolchain). Such wheels would also be compatible with other macOS builds of CPython (such as the ones in Xcode and home-brew). 

Having support for this in this project is a prerequisite for me asking users of setuptools-rust to provide universal2 binary wheels (although there's also the question of CI support, none of the large cloud CI systems currently have Apple M1 runners)

Reply:
I think to set deployment target to macOS 10.9 you just need to set environment variable `MACOSX_DEPLOYMENT_TARGET=10.9`. https://stackoverflow.com/a/64864364